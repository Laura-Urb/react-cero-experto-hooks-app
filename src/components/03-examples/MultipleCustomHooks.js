import React from 'react';
import { useCounter } from '../../hooks/useCounter';
import { useFetch } from '../../hooks/useFetch';
import '../01-useState/counter.css';

export const MultipleCustomHooks = () => {
  const { state, increment } = useCounter(1);
  const { loading, data } = useFetch(
    `https://www.breakingbadapi.com/api/quotes/${state}`
  );
  const { author, quote } = !!data && data[0];
  // (!! la doble negacion convierte null en falso)

  return (
    <div>
      <h1>Multiple Custom Hooks</h1>
      <hr />
      {loading ? (
        <div className='alert alert-info text-center'>Loding...</div>
      ) : (
        <blockquote className='blockquote text-end'>
          <p className='mb-3'>{quote}</p>
          <footer className='blockquote-footer'>{author}</footer>
        </blockquote>
      )}
      <button className='btn btn-primary' onClick={increment}>
        Siguiente Quote
      </button>
    </div>
  );
};
