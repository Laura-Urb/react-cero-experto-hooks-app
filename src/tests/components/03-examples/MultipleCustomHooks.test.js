import { shallow } from 'enzyme';
import { MultipleCustomHooks } from '../../../components/03-examples/MultipleCustomHooks';
import { useCounter } from '../../../hooks/useCounter';
import { useFetch } from '../../../hooks/useFetch';
jest.mock('../../../hooks/useFetch');
jest.mock('../../../hooks/useCounter');

describe('Pruebas MultipleCustomHooks', () => {
  beforeEach(() => {
    useCounter.mockReturnValue({
      state: 10,
      increment: () => {},
    });
  });
  test('debe de mostrarse correctamente ', () => {
    useFetch.mockReturnValue({
      data: null,
      loading: true,
      error: null,
    });
    const wrapper = shallow(<MultipleCustomHooks />);
    expect(wrapper).toMatchSnapshot();
  });
  test('debe de mostrarse la informacion ', () => {
    useFetch.mockReturnValue({
      data: [
        {
          author: 'Juan',
          quote: 'Hello',
        },
      ],
      loading: false,
      error: null,
    });
    const wrapper = shallow(<MultipleCustomHooks />);
    expect(wrapper.find('.alert').exists()).toBe(false);
    expect(wrapper.find('.mb-3').text().trim()).toBe('Hello');
    expect(wrapper.find('footer').text().trim()).toBe('Juan');
  });
});
