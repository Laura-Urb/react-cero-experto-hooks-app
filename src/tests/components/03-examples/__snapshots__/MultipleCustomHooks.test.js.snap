// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas MultipleCustomHooks debe de mostrarse correctamente  1`] = `
<div>
  <h1>
    Multiple Custom Hooks
  </h1>
  <hr />
  <div
    className="alert alert-info text-center"
  >
    Loding...
  </div>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Siguiente Quote
  </button>
</div>
`;
